### PR TITLE
Prevent XSS

### PR DESCRIPTION
**Pull request by Revadike**

### CODE REVIEW

File: src/js/Content/Modules/AugmentedSteam.js [hunk 1]
```diff
@@ -189,7 +189,11 @@ class AugmentedSteam {
             const selector = "a[href*='/linkfilter/']";
 
             for (const node of parent.querySelectorAll(selector)) {
-                node.href = node.href.replace(/^.+?\/linkfilter\/\?url=/, "");
+                const link = node.href.replace(/^.+?\/linkfilter\/\?url=\s*/, "");
+                // eslint-disable-next-line no-script-url
+                if (!link.toLowerCase().startsWith("javascript:")) {
```
Comment (candela97) [inline src/js/Content/Modules/AugmentedSteam.js:194]: Does this line actually trigger eslint? From the documentation, it should only warn when setting a `href` with "javascript:...". I don't think stuff like `document.querySelector("a[href^='javascript:']")` triggers a warning.

Comment (Revadike) [reply src/js/Content/Modules/AugmentedSteam.js:194]: I did for me
¯\_(ツ)_/¯

**Comment by MxtOUT** [comment]
I tried to inject code via a link in a guide, but Steam always prepends the http protocol (so it would be http://javascript:alert(1)).
If you can show me an example where it works, please DM me

**Comment by Revadike** [comment]
https://steamcommunity.com/linkfilter/?url=https://steamcommunity.com/linkfilter/?url=javascript:alert(g_sessionID)

**Comment by MxtOUT** [comment]
I mean a way this would be a valid link you could click on on Steam

**Comment by Revadike** [comment]
I'm not patching xss possible by steam, only made possible with AS.

**Comment by MxtOUT** [comment]
Ok, but the JS is not executed by the link that you sent

**Comment by Revadike** [comment]
Doesn't matter

**Comment by MxtOUT** [comment]
But I asked you to provide an example where you inject arbitrary code

**Comment by Revadike** [comment]
> But I asked you to provide an example where you inject arbitrary code

Ok, I don't have one at hand.

**Pull request closed without merging**